libeos-updater-util/util.c

```diff
@@ -492,3 +493,93 @@ eos_updater_setup_quit_file (const gchar *path,
 
   return g_steal_pointer (&quit_file);
 }
+
+gboolean
+eos_string_to_signed (const gchar  *str,
+                      gint64        min,
+                      gint64        max,
+                      gint64       *out_num,
+                      GError      **error)
+{
+  gint64 number;
+  const gchar *end_ptr = NULL;
+
+  g_return_val_if_fail (str != NULL, FALSE);
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  if (str[0] == '\0')
+    {
+      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                           "empty string is not a number");
+      return FALSE;
+    }
+
+  errno = 0;
+  number = g_ascii_strtoll (str, (gchar **)&end_ptr, 10);
+  if (errno != 0 ||
+      end_ptr == NULL ||
+      *end_ptr != '\0')
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "not a number %s", str);
+      return FALSE;
+    }
+  if (number < min ||
+      number > max)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "number %" G_GINT64_FORMAT " out of bounds (%" G_GINT64_FORMAT " - %" G_GINT64_FORMAT ")",
```

Comment:
Nitpick: Probably better to use a standard [interval format](https://en.wikipedia.org/wiki/Interval_(mathematics)#Notations_for_intervals) in the message.